### PR TITLE
feat: started adding rabbitmq functionality

### PR DESCRIPTION
**Pull request by devanbenz**
- Started adding Queue configuration class 
- Adds docker compose update for rabbitmq

### CODE REVIEW

**Comment by lazyguru** [comment]
I added a comment, but deferring to @jgrim for full review

File: docs/docker-compose.yml [hunk 1]
```diff
@@ -51,6 +51,12 @@ services:
       SUBLINKS_JWT_SECRET: <JWT-SECRET>
       SUBLINKS_BASE_URL: https://demo.sublinks.org
       SUBLINKS_PICTRS_URL: http://pictrs:8080
+      FEDERATION_QUEUE_ENDPOINT: localhost
```
Comment (jgrim) [inline docs/docker-compose.yml:54]: I prefer the nomenclature FEDERATION_QUEUE_HOST rather than FEDERATION_QUEUE_ENDPOINT.

**Review by Pdzly — DISMISSED** [review]
I cant see anything bad. As you already fixed what jgrim marked.

**Comment by devanbenz** [comment]
Added a few changes 
1. Added Json message deserialization to Rabbit listener factory 
2. Modified it so the Producer and Consumer are using different queues (wouldn't make sense for them both to use the same queue 😆 )

**Comment by lazyguru** [comment]
> I added a comment, but deferring to @jgrim for full review

I just now realized my comment was "pending" as I hadn't clicked "start review" instead of "add comment" 🤦🏻. You resolved it already though as it was about you only having one queue and needing to add the second queue for bi-directional communication

File: docs/docker-compose.yml [hunk 2]
```diff
@@ -51,6 +51,13 @@ services:
       SUBLINKS_JWT_SECRET: <JWT-SECRET>
       SUBLINKS_BASE_URL: https://demo.sublinks.org
       SUBLINKS_PICTRS_URL: http://pictrs:8080
+      FEDERATION_QUEUE_HOST: localhost
+      FEDERATION_QUEUE_PORT: 5672
+      FEDERATION_QUEUE_USER: sublinks
+      FEDERATION_QUEUE_PASS: sublinks
+      FEDERATION_QUEUE_NAME: federation # TODO: Need to get correct queue name
+      BACKEND_QUEUE_NAME: backend # TODO: Need to get correct queue name
+      BACKEND_TOPIC_NAME: backend # TODO: Need to get correct topic name
```
Comment (lazyguru) [inline docs/docker-compose.yml:60]: I don't think these need to be TODO items. Using variables here makes it configurable and having the default be backend/federation is probably fine

**Comment by devanbenz** [comment]
> > I added a comment, but deferring to @jgrim for full review
> 
> 
> 
> I just now realized my comment was "pending" as I hadn't clicked "start review" instead of "add comment" 🤦🏻. You resolved it already though as it was about you only having one queue and needing to add the second queue for bi-directional communication

Haha sounds good!

File: docs/docker-compose.yml (hunk 2 above)
Comment (devanbenz) [reply docs/docker-compose.yml:60]: Sounds good will remove comments

**Review by jgrim — APPROVED** [review]

**Pull request merged**